### PR TITLE
UI and tooltip content review

### PR DESCRIPTION
**Pull request by edonehoo**
Edits to content based on PatternFly best practices
https://issues.redhat.com/browse/ISPN-14175

### CODE REVIEW

**Comment by dpanshug** [comment]
Hi, @edonehoo thanks for contributing. Was there any issue raised for this?

**Comment by dpanshug** [comment]
@dvagnero you might be knowing more about this.

**Comment by dvagnero** [comment]
Hi @dpanshug and @karesti! Erin is from the UXD Content Design team and she's performing the UX review of the console text strings. The review is still in progress but feel free to leave your comments. Many thanks @edonehoo

File: src/app/assets/languages/en.json [hunk 1]
```diff
@@ -198,7 +198,7 @@
           "transactional-disabled": "This cache can not be transactional",
           "transactional-disabled-description": "Transactional caches are available in synchronous replication mode only. Remove the capability or chose synchronous replication.",
           "backups-disabled": "Backups are disabled",
-          "backups-disabled-description": "Backups are available only when you configure {{brandname}} to perform cross-site replication. Remove the capability or enable cross-site replication.",
+          "backups-disabled-description": "Backups are available only when you configure {{brandname}} to perform cross-site replication. To add a backup, enable cross-site replication. Otherwise, remove the capability.",
```
Comment (dvagnero) [inline src/app/assets/languages/en.json:201]: It's a complete backup configuration. Users add backup sites, choose backup strategy, merge policy, timeouts, and much more.

File: src/app/assets/languages/en.json [hunk 2]
```diff
@@ -220,11 +220,11 @@
...
           "max-count-helper-invalid": "You must specify a positive number",
           "eviction-strategy": "Eviction strategy",
-          "eviction-strategy-tooltip": "Specify how {{brandname}} removes entries when the cache reaches the threshold that you configure. The REMOVE strategy evicts older entries to make space for new ones. The EXCEPTION strategy configures {{brandname}} to log an exception and prevent new entries from being created.",
+          "eviction-strategy-tooltip": "Specify how {{brandname}} should remove entries when the cache reaches the threshold that you configure. The REMOVE strategy evicts older entries to make space for new ones. The EXCEPTION strategy configures {{brandname}} to log an exception and prevent new entries from being created.",
           "indexed": "Indexing",
-          "indexed-tooltip": "Build an index of values in the cache for faster queries and full-text search capabilities.",
+          "indexed-tooltip": "To enable faster querying and full-text search, build an index of cache values.          ",
```
Comment (dvagnero) [inline src/app/assets/languages/en.json:225]:
```suggestion
          "indexed-tooltip": "To enable faster querying and full-text search, build an index of cache values.",
```

File: src/app/assets/languages/en.json [hunk 3]
```diff
@@ -235,7 +235,7 @@
           "index-startup-mode-tooltip": "Configure how {{brandname}} builds the index.",
           "index-startup-mode-placeholder": "Select a startup mode",
           "secured": "Security authorization",
-          "secured-description": "Restrict access to the cache to protect data. {{brandname}} denies cache operations if users do not have one of the roles you select.",
+          "secured-description": "To protect cache data, restrict user access. When a user does not have one of the roles selected from the menu below, Infinispan denies cache operations.",
```
Comment (dvagnero) [inline src/app/assets/languages/en.json:238]:
```suggestion
          "secured-description": "To protect cache data, restrict user access. When a user does not have one of the roles selected from the menu below, {{brandname}} denies cache operations.",
```

File: src/app/assets/languages/en.json [hunk 4]
```diff
@@ -171,18 +171,18 @@
...
+          "number-owners-tooltip": "The number of owners you select controls how many copies of each entry should exist in a distributed cache. Adding more copies reduces the risk of data loss but also reduces overall cache capacity. The number of nodes in a distributed cache should always be at least one more than the number of owners.",
           "statistics-title": "Statistics",
           "statistics-enable": "Enable statistics for your cache",
           "statistics-disable": "Disable statistics for your cache",
           "encoding-cache-title": "Select an encoding",
-          "encoding-cache-tooltip": "Choose a media type that describes data that you store in the cache. Use ProtoStream encoding for optimal performance and interoperability between client applications.",
+          "encoding-cache-tooltip": "Choose a media type that describes the type of data stored in the cache. Select ProtoStream encoding for optimal performance and interoperability between client applications.",
```
[the remark this answers is not in the document]
Comment (oraNod) [reply src/app/assets/languages/en.json:183]: Select is fine but I don't agree with the change in the first sentence (s/data .../the type of data...). It's too wordy and repetitious.

File: src/app/assets/languages/en.json [hunk 5]
```diff
@@ -171,18 +171,18 @@
           "mode-distributed": "Distributed",
           "mode-replicated": "Replicated",
           "cluster-repl-title": "Configure cluster replication",
-          "cluster-repl-tooltip": "Clusters can replicate data between nodes using synchronous or asynchronous communication. Synchronous replication provides consistency but can slow down write operations. Asynchronous replications gives you better throughput for write operations but can lead to inconsistency and data loss.",
+          "cluster-repl-tooltip": "Clusters can replicate data between nodes using synchronous or asynchronous communication. Synchronous replication provides consistency but can slow down write operations. Asynchronous replication gives you better throughput for write operations, but can lead to inconsistency and data loss.",
```
Comment (oraNod) [inline src/app/assets/languages/en.json:174]: Comma after but is a grammatical error. The second clause is not independent.

File: src/app/assets/languages/en.json [hunk 6]
```diff
@@ -171,18 +171,18 @@
...
           "cluster-repl-async": "Asynchronous",
           "cluster-repl-sync": "Synchronous",
           "number-owners": "Set the number of owners",
-          "number-owners-tooltip": "Controls how many copies of each entry should exist in a distributed cache. Maintaining more replicas of each entry reduces the risk of data loss but also reduces overall capacity. The number of nodes for a distributed cache should always be at least one more than the number of owners.",
+          "number-owners-tooltip": "The number of owners that you select controls how many copies of each entry should exist in a distributed cache. Adding more copies reduces the risk of data loss but also reduces overall capacity of nodes during rebalancing operations. The number of nodes for a distributed cache should always be at least one more than the number of owners.",
```
Comment (oraNod) [inline src/app/assets/languages/en.json:178]: "of nodes during rebalancing operations" is not correct. Increasing the number of replicas reduces overall cluster capacity.

File: src/app/assets/languages/en.json [hunk 7]
```diff
@@ -210,7 +210,7 @@
           "cache-feature-list-placeholder": "Select capabilities",
           "cache-feature-list-typeahead": "Select capabilities",
           "bounded": "Bounded",
-          "bounded-tooltip": "Control the size of the cache by configuring {{brandname}} to evict entries. You can configure an eviction threshold based on the maximum amount of memory that a cache can use or based on the total number of entries that a cache can hold.",
+          "bounded-tooltip": "To limit the size of the cache, configure {{brandname}} to evict entries. You can set an eviction threshold based on either the cache’s maximum amount of memory or based on the total number of entries that it can hold.",
```
Comment (oraNod) [inline src/app/assets/languages/en.json:213]: Could say "to restrict" instead of "to limit". Also we've never used possessive with the word cache to avoid anthropomorphism. I think saying "based on the maximum amount of memory or total number of entries" is fine.

File: src/app/assets/languages/en.json [hunk 8]
```diff
@@ -220,11 +220,11 @@
           "max-count-tooltip": "Configure the total number of entries that the cache can contain. When the cache reaches this threshold, {{brandname}} removes entries according to the eviction strategy that you configure.",
           "max-count-helper-invalid": "You must specify a positive number",
           "eviction-strategy": "Eviction strategy",
-          "eviction-strategy-tooltip": "Specify how {{brandname}} removes entries when the cache reaches the threshold that you configure. The REMOVE strategy evicts older entries to make space for new ones. The EXCEPTION strategy configures {{brandname}} to log an exception and prevent new entries from being created.",
+          "eviction-strategy-tooltip": "Specify how {{brandname}} should remove entries when the cache reaches the threshold that you configure. The REMOVE strategy evicts older entries to make space for new ones. The EXCEPTION strategy configures {{brandname}} to log an exception and prevent new entries from being created.",
```
Comment (oraNod) [inline src/app/assets/languages/en.json:223]: Why "should"?

File: src/app/assets/languages/en.json [hunk 9]
```diff
@@ -220,11 +220,11 @@
...
           "indexed": "Indexing",
-          "indexed-tooltip": "Build an index of values in the cache for faster queries and full-text search capabilities.",
+          "indexed-tooltip": "To enable faster querying and full-text search, build an index of cache values.",
```
Comment (oraNod) [inline src/app/assets/languages/en.json:225]: "To enable faster queries"

File: src/app/assets/languages/en.json [hunk 10]
```diff
@@ -235,7 +235,7 @@
           "index-startup-mode-tooltip": "Configure how {{brandname}} builds the index.",
           "index-startup-mode-placeholder": "Select a startup mode",
           "secured": "Security authorization",
-          "secured-description": "Restrict access to the cache to protect data. {{brandname}} denies cache operations if users do not have one of the roles you select.",
+          "secured-description": "To protect cache data, restrict user access. When a user does not have one of the roles selected from the menu below, {{brandname}} denies cache operations.",
```
Comment (oraNod) [inline src/app/assets/languages/en.json:238]: I'm not sure how this is better?

File: src/app/assets/languages/en.json [hunk 11]
```diff
@@ -349,7 +349,7 @@
           "striping": "Lock striping",
           "striping-tooltip": "Uses a shared pool of locks for all entries in the cache. Striping lowers the memory footprint for locks but can reduce concurrency. If you disable striping, a lock is created for each entry in the cache.",
           "index-tuning": "Index tuning",
-          "index-tuning-tooltip": "Control how {{brandname}} interacts with an index to optimize performance.",
+          "index-tuning-tooltip": "To optimize cache performance, specify the methods that {{brandname}} uses to interact with the index.",
```
Comment (oraNod) [inline src/app/assets/languages/en.json:352]: I get that you're flipping sentences to put emphasis on the perceived value or user motivation but I think it's adding unnecessary verbiage.

File: src/app/assets/languages/en.json (hunk 1 above)
Comment (edonehoo) [reply src/app/assets/languages/en.json:201]: Ah, I see. Rolling this back in my next commit, then!

File: src/app/assets/languages/en.json [hunk 12]
```diff
@@ -289,17 +289,17 @@
...
           "full-xa": "FULL_XA",
           "full-xa-tooltip": "{{brandname}} uses transaction recovery and enlists the cache with the Java transaction manager through the XAResource API. Using recovery provides durability by marking transactions as in-doubt when they fail. FULL_XA can impact performance by reducing cluster capacity when storing in-doubt transactions and by increasing traffic for failed transaction notifications.",
           "locking-mode": "Locking mode",
-          "locking-mode-tooltip": "Configure how {{brandname}} locks keys to perform write operations for transactions. Locking keys adds contention that increases latency for write operations. You can adjust the amount of contention by using optimistic or pessimistic locking.",
+          "locking-mode-tooltip": "To determine the way that {{brandname}} performs write operations for transactions, select the locking mode below. ",
```
Comment (oraNod) [inline src/app/assets/languages/en.json:302]: This change loses meaning imo.

File: src/app/assets/languages/en.json [hunk 13]
```diff
@@ -394,7 +394,7 @@
           "recovery-cache": "Recovery cache",
           "recovery-cache-tooltip": "Specifies the cache that stores information to recover in-doubt transactions.",
           "transactional-tuning": "Transaction tuning",
-          "transactional-tuning-tooltip": "Configures settings for transactional caches."
+          "transactional-tuning-tooltip": "Configure and adjust settings for transactional caches."
```
Comment (oraNod) [inline src/app/assets/languages/en.json:397]: I think either "Configure settings" or "Adjust settings" is good. Maybe "adjust" because it's more of a synonym with "tune".

File: src/app/assets/languages/en.json (hunk 11 above)
Comment (edonehoo) [reply src/app/assets/languages/en.json:352]: Yes, I made these changes based on [PatternFly best practices](https://www.patternfly.org/v4/ux-writing/ux-writing-best-practices#lead-with-the-benefit), and how my team generally writes. But, of course, these suggestions are optional. Because I don't have full context on this tool, you all will know better than I do if they don't add value or if they make the interface more confusing. These can easily be left unchanged if that's what you'd like!

File: src/app/assets/languages/en.json (hunk 8 above)
Comment (edonehoo) [reply src/app/assets/languages/en.json:223]: My interpretation was that eviction "strategy" was a choice that the user was making, and that "should" reflected that more. But this doesn't necessarily add new value if you prefer it without.

**Comment by edonehoo** [comment]
I've reviewed everything, so I don't have any new feedback to add at this point. If any of the changes that I suggested aren't desirable, then I can roll back my edits as I've done in a few instances. Just point them out and let me know!

File: src/app/assets/languages/en.json (hunk 11 above)
Comment (oraNod) [reply src/app/assets/languages/en.json:352]: Thanks for linking to the PatternFly best practices @edonehoo

File: src/app/assets/languages/en.json (hunk 10 above)
Comment (oraNod) [inline src/app/assets/languages/en.json:238]:
```suggestion
          "secured-description": "To protect data, restrict user access to the cache. When a user does not have one of the roles selected from the menu below, {{brandname}} denies cache operations.",
```

File: src/app/assets/languages/en.json (hunk 6 above)
Comment (dvagnero) [reply src/app/assets/languages/en.json]: Thanks Don. This one is on me

**Comment by dvagnero** [comment]
> Hi, @edonehoo thanks for contributing. Was there any issue raised for this?

Do we need an ISPN issue to track this?

**Comment by dpanshug** [comment]
> > Hi, @edonehoo thanks for contributing. Was there any issue raised for this?
> 
> Do we need an ISPN issue to track this? @dpanshug

Yes, would be useful to track.

**Comment by dvagnero** [comment]
@edonehoo Could you please rename your branch to ISPN-14175?

**Comment by dvagnero** [comment]
@dpanshug could you please review the PR?
